en/docs/integration-guides/usecases/datamapper/read-csv-file-and-transform-to-xml-file.md

```diff
@@ -0,0 +1,601 @@
+
+    - **Name:** xmlRecord
+
+    This variable name represents the XML output structure that will hold the mapped data.
+
+
+  3. Assign the Variable Type
+
+    1. Click the **Type** field (marked with the blue `T` icon).
+    2. Begin typing `Order`.
+    3. From the dropdown suggestions, select **Orders**.
+
+    This binds the variable to your predefined `Orders` record type, which defines the XML schema structure for the output.
+
+
+  4. Open the Data Mapper
+
+    Once the variable name and type are defined, click **Open in Data Mapper** at the bottom right of the properties pane.
+
+    This action opens the **Data Mapper** where you can map CSV fields to XML elements visually.
+
+
+  5. Explore the Data Mapper Interface
+
+    Inside the Data Mapper view, you’ll see two main panels:
+
+    **Left Panel — Input Data**
+
+    - `csvRecords` → Type: `CSV[]`
+    - `inputCSV` → Type: `string`
+    - `outputXML` → Type: `string`
+
+    **Right Panel — Output Structure**
+
+    - `xmlRecord` → Type: `Orders`
+    - `Row[]`
+        - `index` (int)
+        - `order_id` (string)
+        - `sku` (string)
+        - `qty` (int)
+        - `price` (decimal)
+
+    This view provides a visual workspace to connect input CSV fields with XML output fields directly.
+
+
+At this point, you have:
+
+1. Declared a new variable named *xmlRecord* of type `Orders`.
+2. Opened it inside the Data Mapper.
+3. Verified that both `csvRecords` (input) and `xmlRecord` (output) are available for mapping.
+
+You’re now ready to perform the visual mapping between CSV and XML data structures.
+
+<a href="{{base_path}}/assets/usecases/datamapping/csv-to-xml-simple-mapping/img/csv_to_xml6.gif"><img src="{{base_path}}/assets/usecases/datamapping/csv-to-xml-simple-mapping/img/csv_to_xml6.gif" alt="Read the csv file" width="70%"></a>
+
+
+## Step 9: Map csv fields to xml rows in the data mapper
+
+Now that you’ve opened the Data Mapper view, it’s time to visually connect the fields between your CSV input and XML output. This step transforms the tabular data structure into a hierarchical XML format.
+
+
+  1. Understand the Mapping Context
+
+      On the **left**, you have:
+      - `csvRecords` → Type: `CSV[]`
+          - Each item contains: `order_id`, `sku`, `qty`, and `price` (all as `string`).
+
+      On the **right**, you have:
+      - `xmlRecord` → Type: `Orders`
+          - Contains: `Row[]`
+              - Each `Row` has: `index (int)`, `order_id (string)`, `sku (string)`, `qty (int)`, and `price (decimal)`.
+
+
+  2. Start the Mapping by Linking the Record Arrays
+
+      1. Drag from `csvRecords` (left) to `Row` (right).
+      2. This creates a parent-level mapping between the two collections.
+
+    If you see an error such as:
+    > `incompatible types: expected 'Row[]', found 'CSV[]'`
+
+    That’s expected — the data mapper is warning that nested mapping is required to handle field-level conversions.
+
+
+  3. Map Individual Fields
```

Comment:
The `Map with Query Expression` step is missing.

i.e. 
<img width="1120" height="606" alt="image" src="https://github.com/user-attachments/assets/020726c4-e72a-4219-8155-f464d7ca04fc" />